File: src/main/java/cn/chahuyun/economy/entity/props/Equipment.java
package cn.chahuyun.economy.entity.props;

import cn.chahuyun.economy.entity.UserInfo;
import cn.chahuyun.economy.prop.PropBase;

/**
 * 装备
 *
 * @author Moyuyanli
 * @date 2024-10-15 11:35
 */
public class Equipment extends PropBase {
    /**
     * 商店显示描述
     *
     * @return 商店显示结果
     */
    @Override
    public String toShopInfo() {
        return null;
    }

    /**
     * 使用该道具
     *
     * @param user
     */
    @Override
    public void use(UserInfo user) {

    }
}

File: src/main/kotlin/cn/chahuyun/economy/model/props/Equipment.kt
package cn.chahuyun.economy.model.props

import cn.chahuyun.economy.prop.AbstractProp
import cn.chahuyun.economy.prop.Usable
import cn.chahuyun.economy.prop.UseResult

/**
 * 装备 (Kotlin 重构版)
 */
class Equipment(
    kind: String = "equipment",
    code: String = "",
    name: String = "",
) : AbstractProp(kind, code, name), Usable {

    override fun use(event: UseEvent): UseResult {
        // 装备逻辑：比如穿上装备，增加属性等
        return UseResult.success("已装备 $name")
    }

    override fun toShopInfo(): String {
        return "装备名称: $name\n装备描述: $description\n装备价值: $cost 金币"
    }
}

